controller/pkg/ipsetmanager/ipsets.go

```diff
@@ -0,0 +1,363 @@
+package ipsetmanager
+
+import (
+	"encoding/base64"
+	"io"
+	"net"
+	"sync"
+
+	ipsetpackage "github.com/aporeto-inc/go-ipset/ipset"
+	"github.com/spaolacci/murmur3"
+	provider "go.aporeto.io/trireme-lib/controller/pkg/aclprovider"
+
+	"go.aporeto.io/trireme-lib/policy"
+	"go.uber.org/zap"
+)
+
+const (
+	//IPv6DefaultIP is the default ip of v6
+	IPv6DefaultIP = "::/0"
+	//IPv4DefaultIP is the  default ip for v4
+	IPv4DefaultIP = "0.0.0.0/0"
+	//IPsetV4 version for ipv4
+	IPsetV4 = iota
+	//IPsetV6 version for ipv6
+	IPsetV6
+)
+
+type ipsetInfo struct {
+	contextIDs map[string]bool
+	name       string
+	addresses  map[string]bool
+}
+
+type handler struct {
+	serviceIDtoIPset      map[string]*ipsetInfo
+	contextIDtoServiceIDs map[string]map[string]bool
+	ipset                 provider.IpsetProvider
+	ipsetPrefix           string
+	ipFilter              func(net.IP) bool
+	ipsetParams           *ipsetpackage.Params
+	toDestroy             []string
+}
+
+var lock sync.RWMutex
+var ipv4Handler *handler
+var ipv6Handler *handler
+
```

Comment:
Let's not use package wide variables. It always leads to a mess. Create an object and pack the variables in the object and initialize the object.